src/zkp/surjection_proof.rs

```diff
@@ -187,6 +188,17 @@ impl ::core::fmt::Display for SurjectionProof {
     }
 }
 
+impl std::str::FromStr for SurjectionProof {
+    type Err = Error;
+    fn from_str(s: &str) -> Result<SurjectionProof, Error> {
+        let mut res = Vec::new();
+        match from_hex(s, &mut res) {
+            Ok(_) => SurjectionProof::from_slice(&res),
+            _ => Err(Error::InvalidSurjectionProof),
+        }
+    }
+}
+
 // TODO: Macrofy (de)serialization
```

Comment:
I think this can be deleted now, I would consider the impl to be concise enough.